extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/YoutubeChannelExtractorTest.java

```diff
@@ -648,4 +649,100 @@ public void testVerified() throws Exception {
             assertFalse(extractor.isVerified());
         }
     }
+
+    /**
+     * Test the extraction of channel data from a RichGridRenderer (currently A/B tested, as of 11.10.2022)
+     */
+    public static class RichGrid implements BaseChannelExtractorTest {
+        private static YoutubeChannelExtractor extractor;
+
+        @BeforeAll
+        public static void setUp() throws Exception {
+            YoutubeTestsUtils.ensureStateless();
+            NewPipe.init(DownloaderFactory.getDownloaderOfType(RESOURCE_PATH + "richGrid", DownloaderType.MOCK));
```

Comment:
Why is the mock downloader needed here?

Reply:
Because the new layout is A/B tested, I had to use the mock downloader to reproduce it.

In my channel tabs PR I added the option to manually set the visitor data cookie which allows you to reproduce A/B tests without using the mock downloader.